### DISCUSSION TITLE
Support skipping git authorization check

### DISCUSSION THREAD

**Discussion by Blackclaws** [semantic-release/semantic-release#2581] [Release workflow]
### New feature motivation

I am using semantic-release as part of a local development workflow where I am determining the potential new version of a package in order to adjust the package version requested by other projects that depend on it.

It works fine up to the point where you try to go offline or simply don't have push permissions (albeit you have pull permissions) to one of the repositories involved.

I'd like to be able to skip the check for git authorization because it simply isn't relevant to the use case as no release is being made and I am using semantic-release purely to determine the next version it would produce in the actual setting.

According to the documentation this feature is in place mainly to help with debugging. However users that use it in such an advanced capacity might want to disable this functionality in certain cases.

### New feature description

Introduce a command line switch --no-git-verify/--skip-git-verify that disables the verification behavior for git.

**Comment by travi** [comment]
> I am determining the potential new version of a package in order to adjust the package version requested by other projects that depend on it

please explain your use case more thoroughly. i dont understand what your goal really is with this

**Reply by Blackclaws** [reply]
I am developing a toolkit that puts a number of different repos that might have dependencies between them into a local workspace where development can happen simultaneously. I'd like for the repos that have dependencies on another repo to have their package.json or equivalent for other languages point to the new version that their dependency will get once it is actually published.

The idea is that the toolset would allow you to start a merge request for all of the edited repos simultaneously. Of course you'd still have to merge them in order, however what I would like to do is adjust the version of their respective dependencies that they are referring to to the version that they will most likely be in.

An example:

Let's say I have two repos: A and B. 

B depends on A

A starts of as 1.0.0
B starts of as 1.0.0

I now start local development on B and A. After a number of commits I am happy with the result. However I do want to put a merge request up for both of them now. I already know that A has had some features added, so its version will definitely increase. I would now like to put B into the state where it already references the future version of A before I push it, therefore skipping (in the best case that no-one else has merged something in the meantime) another round of commits to B that I would have to do once A has been published.

Since we are using semantic-release to actually perform the releases, it seems only natural to also use semantic-release to determine the future version. This works fine in online mode and if all permissions are available, however if I do not have permissions to for example push to the main branch of A or I am working in an offline environment, I cannot determine the future version of A because semantic-release will fail at the stage where it tries to validate the push to git.
